ext/dom/tests/gh17145.phpt

```diff
@@ -0,0 +1,18 @@
+--TEST--
+GH-17145 (DOM memory leak)
+--EXTENSIONS--
+dom
+--CREDITS--
+YuanchengJiang
+--FILE--
+<?php
+$element = new DOMElement("N", "W", "y");
+$attr = new DOMAttr("c" , "n");
+$doc = new DOMDocument();
+$doc->appendChild($element);
+$element->setAttributeNodeNS($attr);
+$attr->appendChild($doc->createEntityReference('amp'));
+echo $attr->value;
+?>
+--EXPECT--
+n&
```

Comment:
Looking at CI, is this `&` difference coming from libxml2 versions ?

Reply:
Looks like it :( It seems that the CI jobs using a more recent version of libxml have the correct output while the older versions do not. I'll run a bisect on libxml to see what commit in libxml fixed this.

Reply:
This was fixed in https://github.com/GNOME/libxml2/commit/b8597f46df99b7a819e2c14ea1c24f649c0b149e, so it's a pretty recent fix and since distros are quite behind on libxml updates, it'll take some time for them to get this fix.
It would be possible to patch this in PHP too, but I'm going for a SKIPIF right now as I don't want to add more workarounds than we already have to be honest...

Reply:
Not really happy about dealing with these kinds of stuff